README.md

```diff
@@ -1,92 +1,192 @@
 
 Add to your global rebar3 config in `~/.config/rebar3/rebar.config`:
 
 ```erlang
 {plugins, [rebar3_hex]}.
 ```
 
-Usage
---------
+Alternatively, you can add to your project's rebar3 config in project plugins : 
 
-## Authenticating User
+```erlang
+{project_plugins, [rebar3_hex]}.
+```
 
-If you already have a user for [hex.pm](https://hex.pm) run:
+**NOTE** : Be sure **not** to add `rebar3_hex` to the plugins section within a projects rebar3 config as this will 
+become a dependency for others downloading your package from hex.pm
 
+### Documentation
 
-```shell
-$ rebar3 hex user auth
-```
+rebar3_hex by default expects you to configure a documentation provider. We recommend using
+[rebar3_ex_doc](https://hexdocs.pm/rebar3_ex_doc/) for publishing documentation along with your package for a 
+consistent format and style on [hex.pm](https://hex.pm/). 
+
+Example : 
 
-Note that this will ask for your hex.pm username and password, as well as a password for encrypting your api token that has write permissions to the repository. When publishing a package you will have to give this password to decrypt the token in order to publish.
+```erlang
 
-## Private Organizations
+{ex_doc, [
+    {source_url, <<"https://github.com/namespace/your_app">>},
+    {extras, [<<"README.md">>, <<"LICENSE">>]},
+    {main, <<"readme">>}
+]}.
 
-[Private organizations]() are treated as repositories that have a parent. The parent is found as the first part of a repository's name, separated from the organization by a `:`. So for an organization `rebar3` on the main repository `hexpm` the name must be `hexpm:rebar3`.
+{hex, [{doc, ex_doc}]}.
+```
 
-`~/.config/rebar3/rebar.config`
+Alternatively you can use the edoc provider that ships with rebar3 : 
 
 ```erlang
-{hex, [{repos, [
-		#{name => <<"hexpm:rebar3">>}
-	  ]}]}.
+{hex, [{doc, edoc}]}.
 ```
 
-## Read-Only Repo Key for CI
+## Basic usage 
+
+You should read the [docs](https://hexdocs.pm/rebar3_hex/) for a complete over for each provider, but below is a
+brief overview of basic usage. 
+
+### Authenticating User
+
+If you already have a user for [hex.pm](https://hex.pm) run:
 
-If you have a private organization or other private repository it is recommended that you use a repo specific auth token for reading from the repository in CI. To generate a token:
 
 ```shell
-$ rebar3 hex repo auth <repo> generate
-Generated key: abc123
+$ rebar3 hex user auth
 ```
 
-Then in CI use whatever method is available for setting an environment variable to the token and add this call at the beginning of your CI runs to add the token to your rebar3 hex tokens:
+Note that this will ask for your hex.pm username and password, as well as a password for encrypting your api token that 
+has write permissions to the repository. When publishing a package you will have to give this password to decrypt the 
+token in order to publish.
+
+See the [docs](https://hexdocs.pm/rebar3_hex/rebar3_hex_user.html) for more information.
```

Comment:
Not a working link.

Also: https://hex.pm/docs/rebar3_publish and others may need updates?

Reply:
Right, these links won't work until v7 is published. Will have to do a PR to hex.pm and rebar3.org as well. Chicken and egg problem though, thoughts?